src/node_webstorage.cc

```diff
@@ -0,0 +1,661 @@
+  }
+}
+
+static void IndexedStorageGetter(uint32_t index,
+                                 const PropertyCallbackInfo<Value>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  StorageGetter(Uint32ToName(env->context(), index), info);
+}
+
+static void IndexedStorageSetter(uint32_t index,
+                                 Local<Value> value,
+                                 const PropertyCallbackInfo<Value>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  StorageSetter(Uint32ToName(env->context(), index), value, info);
+}
+
+static void IndexedStorageQuery(uint32_t index,
+                                const PropertyCallbackInfo<Integer>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  StorageQuery(Uint32ToName(env->context(), index), info);
+}
+
+static void IndexedStorageDefiner(uint32_t index,
+                                  const PropertyDescriptor& desc,
+                                  const PropertyCallbackInfo<Value>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  StorageDefiner(Uint32ToName(env->context(), index), desc, info);
+}
+
+static void IndexedStorageDeleter(uint32_t index,
+                                  const PropertyCallbackInfo<Boolean>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  StorageDeleter(Uint32ToName(env->context(), index), info);
+}
+
+static void StorageLengthGetter(const FunctionCallbackInfo<Value>& info) {
+  Storage* storage;
+  ASSIGN_OR_RETURN_UNWRAP(&storage, info.This());
+  info.GetReturnValue().Set(storage->Length());
+}
+
+static void StorageLengthSetter(const FunctionCallbackInfo<Value>& info) {
+  info.GetReturnValue().SetUndefined();
+}
+
+static void Initialize(Local<Object> target,
+                       Local<Value> unused,
+                       Local<Context> context,
+                       void* priv) {
+  Environment* env = Environment::GetCurrent(context);
+  Isolate* isolate = env->isolate();
+  auto ctor_tmpl = NewFunctionTemplate(isolate, Storage::New);
+  auto inst_tmpl = ctor_tmpl->InstanceTemplate();
+
+  inst_tmpl->SetInternalFieldCount(Storage::kInternalFieldCount);
+  inst_tmpl->SetHandler(NamedPropertyHandlerConfiguration(
+      StorageGetter,
+      StorageSetter,
+      StorageQuery,
+      StorageDeleter,
+      StorageEnumerator,
+      StorageDefiner,
+      nullptr,
+      Local<Value>(),
+      PropertyHandlerFlags::kHasNoSideEffect));
+  inst_tmpl->SetHandler(IndexedPropertyHandlerConfiguration(
+      IndexedStorageGetter,
+      IndexedStorageSetter,
+      IndexedStorageQuery,
+      IndexedStorageDeleter,
+      nullptr,
+      IndexedStorageDefiner,
+      nullptr,
+      Local<Value>(),
+      PropertyHandlerFlags::kHasNoSideEffect));
+
+  Local<FunctionTemplate> length_getter = FunctionTemplate::New(
+      isolate, StorageLengthGetter, Local<Value>());
+  Local<FunctionTemplate> length_setter = FunctionTemplate::New(
+      isolate, StorageLengthSetter, Local<Value>());
```

Comment:
Nit: the `Local<Value>()` on these are unnecessary, aren't they? I believe these default.